### ISSUE TITLE
Milliseconds is 0 in the timestamp after kruize restart

### ISSUE THREAD

**Issue by chandrams** [kruize/autotune#774]
Milliseconds is 0 in the timestamp after kruize restart

cat list_reco_json_before_1.json | grep monitoring_end_time | sort | uniq
  "monitoring_end_time": "2023-04-23T22:02:04.659Z",
  "monitoring_end_time": "2023-04-23T22:17:04.659Z",
  "monitoring_end_time": "2023-04-23T22:32:04.659Z",
  "monitoring_end_time": "2023-04-23T22:47:04.659Z",
 
cat list_reco_json_after_1.json | grep monitoring_end_time | sort | uniq
"monitoring_end_time": "2023-04-23T22:02:04.000Z",
"monitoring_end_time": "2023-04-23T22:17:04.000Z",
"monitoring_end_time": "2023-04-23T22:32:04.000Z",
"monitoring_end_time": "2023-04-23T22:47:04.000Z",

**Comment by chandrams** [comment]
This I think should be fixed in PR 756 as per slack communication, raising it to keep track to verify it once the build is available

**Comment by chandrams** [comment]
Verified with kruize/autotune_operator:0.0.13_rm, I see the below differences

cat list_reco_json_before_1.json | grep monitoring_end_time | sort | uniq
                                            "monitoring_end_time": "2023-04-24T16:50:04.170Z",
                                            "monitoring_end_time": "2023-04-24T17:05:04.170Z",
                                            "monitoring_end_time": "2023-04-24T17:20:04.170Z",
                                            "monitoring_end_time": "2023-04-24T17:35:04.170Z",
 cat list_reco_json_after_1.json | grep monitoring_end_time | sort | uniq
                                            "monitoring_end_time": "2023-04-24T16:50:04.017Z",
                                            "monitoring_end_time": "2023-04-24T17:05:04.017Z",
                                            "monitoring_end_time": "2023-04-24T17:20:04.017Z",
                                            "monitoring_end_time": "2023-04-24T17:35:04.017Z",

**Comment by chandrams** [comment]
```
        "data": {
                                "2023-04-24T16:50:04.170Z": {
                                    "duration_based": {
                                        "short_term": {
                                            "monitoring_start_time": "2023-04-23T16:50:04.170Z",
                                            "monitoring_end_time": "2023-04-24T16:50:04.170Z",
                                            "duration_in_hours": 24.0,
```
After kruize restart

```
    "2023-04-24T16:50:04.170Z": {
                                    "duration_based": {
                                        "short_term": {
                                            "monitoring_start_time": "2023-04-23T16:50:04.017Z",
                                            "monitoring_end_time": "2023-04-24T16:50:04.017Z",
                                            "duration_in_hours": 24.0,
```

**Comment by dinogun** [comment]
Tested with these timestamps and it seems to be working fine
https://github.com/kruize/kruize-demos/pull/57/files

**Comment by bharathappali** [comment]
@chandrams Can we close this if you are comfortable with the latest release fix?

**Comment by chandrams** [comment]
@bharathappali - This is an intermittent issue or one of case, if you can also try out we can confirm. Thanks

**Comment by bharathappali** [comment]
Sure @chandrams Will try out and confirm you. Thanks for the jsons & demo script zip.

**Comment by bharathappali** [comment]
Fixed in #793

**Comment by chandrams** [comment]
Verified with kruize/autotune_operator:0.0.14_mvp, issue is fixed now.
